pkg/datapath/loader/compile.go

```diff
@@ -284,6 +290,9 @@ func compile(ctx context.Context, prog *progInfo, dir *directoryInfo) (err error
 
 	args = append(args, standardCFlags...)
 	args = append(args, progCFlags(prog, dir)...)
+	if len(prog.ExtraCArgs) != 0 {
+		args = append(args, prog.ExtraCArgs...)
```

Comment:
Also, is it worth coming up with a more specific name for this flags, instead of stuffing them under "extra" args? The main reason is for maintainability and knowing which args are being passed to the compiler and when. I can see "extra" being potentially abused in the future, where people continue to stuff more and more stuff in it.

Reply:
Hmm, not sure what better name would be? It's really just Extra C(compiler) Args of a given program.

Alternative approach would have been to put some XDP compilation logic into `compile()`, but the function is generic, so I didn't want to trash it with the custom logic.

Reply:
Fair enough